docs/docs/behaviors/macros.md

```diff
@@ -109,6 +109,23 @@ bindings
     ;
 ```
 
+### Processing Continuation on Layer Change
+
+The macro can also be paused so that part of the `bindings` list is processed when the layer changes.
```

Comment:
I think being more specific here will be nice: is it watching for the layer that the macro is on to be turned off? Or any change in the currently active set of layers?

Reply:
Right now it's the latter: any change in the active set of layers will release all scheduled states. The former could be nice, but it would add some complexity to allow for storing scheduled states from different layers and releasing only some of them at a time.

Reply:
Question: if I wanted to get the layer the macro was in, would I store the `zmk_behavior_binding_event->layer` int passed into `on_macro_binding_pressed()`?

And then to check if the layer is turned off in `macro_layer_state_changed_listener()`, see if any bit in the stored `layer` int has been switched off?

Reply:
I added a commit that enables the first behavior you suggested - it now watches for the layer the macro is on to be turned off. I also edited the documentation to be more specific.

Reply:
I think that's intuitive but either way could have been fine as long as it is documented.

One thing I am curious is how it behaves when the macro is not invoked directly from a layer binding, like
- a combo
- nested in another behavior (hold-tap, mod-morph, another macro etc.)

I am not sure it needs to work with combos since it doesn't make much sense, but maybe a warning note will be warranted.

Reply:
Yeah, I added that behavior because I thought it would be more intuitive as well, so that adding another layer wouldn't trigger all continuations. However, it does limit `macro_pause_for_layer` to be used only on non-default layers.

I just tested with a combo and it seems to count as layer 0 even if a different layer is active at the time, so it would not work with the current behavior.

Perhaps one way a combo would make sense is where app switch could be repeatedly activated with a combo in the base layer, and momentarily entering any other layer would then exit the app switcher. Not the way I'm used to using Alt+Tab, but perhaps useful for someone who is averse to holding multiple keys.

This use would require going back to the original behavior of continuing on _any_ layer change, rather than waiting for the current layer's deactivation.